### ISSUE TITLE
Dependency missing: libtiff.so.5

### ISSUE THREAD

**Issue by diegoweb** [remicollet/remirepo#109]
Trying to install php72 with REMI-PHP72 and REMI-SAFE enabled.

Run: yum update

```
[...]
---> Package php-xmlrpc.x86_64 0:5.4.16-45.el7 will be updated
---> Package php-xmlrpc.x86_64 0:7.2.11-1.el7.remi will be an update
--> Running transaction check
---> Package gd-last.x86_64 0:2.2.5-5.el7.remi will be installed
--> Processing Dependency: libtiff.so.5(LIBTIFF_4.0)(64bit) for package: gd-last-2.2.5-5.el7.remi.x86_64
--> Processing Dependency: libtiff.so.5()(64bit) for package: gd-last-2.2.5-5.el7.remi.x86_64
---> Package libargon2.x86_64 0:20161029-2.el7 will be installed
---> Package php-fedora-autoloader.noarch 0:1.0.0-1.el7 will be installed
---> Package php-json.x86_64 0:7.2.11-1.el7.remi will be installed
--> Finished Dependency Resolution
Error: Package: gd-last-2.2.5-5.el7.remi.x86_64 (remi-safe)
           Requires: libtiff.so.5()(64bit)
Error: Package: gd-last-2.2.5-5.el7.remi.x86_64 (remi-safe)
           Requires: libtiff.so.5(LIBTIFF_4.0)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

OS: CentOS Linux 7.5.1804

**Comment by remicollet** [comment]
So, "base" and "updates" (default CentOS repositories) are missing

**Comment by diegoweb** [comment]
Thank you! It worked.
I was following a tutorial in Virtualmin forum and the user said to disable all repos but Remi and Epel.
https://www.virtualmin.com/node/48280

Thank you for your help!

**Comment by remicollet** [comment]
> I was following a tutorial in Virtualmin forum and the user said to disable all repos but Remi and Epel.

No, this howto say

> Disable any **extra** repository you have installed...

**extra** being the important word ;)

**Comment by diegoweb** [comment]
Since english is not my native language, I thought this "Disable any extra repository you have installed" meant: "Disable any repository YOUR system have installed".

In my native language, this phrase can have this both meanings =/
You in this case could mean the user and the system (again, in my native language).
My bad.

Thank you!

**Comment by remicollet** [comment]
:)

Welcome

**Comment by craigahumphreys** [comment]
I have same issue, but I have CentOS-Base.repo base and updates enabled
CentOS 6.9.  Consistently get a few dozen broken dependencies such as diegoweb had when attempting to upgrade PHP.

**Comment by craigahumphreys** [comment]
Yeah, that isn't it.  I have default CentOS repositories.  I disabled all others, then added epel and remi. yum install fails every time with extensive errors.  I gave up and started building from source.

**Comment by remicollet** [comment]
Please post full you output, can help without any information.

Mandatory repositories are base, updates, EPEL,  remi-safe and remi-php##

**Comment by craigahumphreys** [comment]
Here are the errors I'm getting and the repositories that are defined in /etc/yum.repo.d:

**Repos**:
- CenOS-Base:  base and upgrade enabled
- epel.repo
- remi-safe.repo
- remi-php72.repo

**Error information when issuing "yum update" command**:
(... a bunch of package requirement analysis output ...)
--> Finished Dependency Resolution
Error: Package: gd-last-2.2.5-5.el7.remi.x86_64 (remi-safe)
           Requires: libpng15.so.15()(64bit)
Error: Package: php-cli-7.2.13~RC1-1.el6.remi.x86_64 (remi-php72-test)
           Requires: libargon2.so.0()(64bit)
Error: Package: gd-last-2.2.5-5.el7.remi.x86_64 (remi-safe)
           Requires: libtiff.so.5()(64bit)
Error: Package: gd-last-2.2.5-5.el7.remi.x86_64 (remi-safe)
           Requires: libpng15.so.15(PNG15_0)(64bit)
Error: Package: php-7.2.13~RC1-1.el6.remi.x86_64 (remi-php72-test)
           Requires: libargon2.so.0()(64bit)
Error: Package: remi-release-7.5-2.el7.remi.noarch (remi-safe)
           Requires: epel-release = 7
           Installed: epel-release-6-8.noarch (@extras)
               epel-release = 6-8
Error: Package: gd-last-2.2.5-5.el7.remi.x86_64 (remi-safe)
           Requires: libwebp.so.4()(64bit)
Error: Package: gd-last-2.2.5-5.el7.remi.x86_64 (remi-safe)
           Requires: libtiff.so.5(LIBTIFF_4.0)(64bit)
Error: Package: gd-last-2.2.5-5.el7.remi.x86_64 (remi-safe)
           Requires: libc.so.6(GLIBC_2.14)(64bit)

**Comment by remicollet** [comment]
Obviously you are missing **el6** and **el7** packages.

Check used version of centos (/Etc/centos-release) and install proper repository.

For memory => https://rpms.remirepo.net/wizard/

**Comment by craigahumphreys** [comment]
yum repolist showed I didn't have anything missing
repo id repo name status
base CentOS-6 - Base 6,713
centosplus CentOS-6 - Plus 34
contrib CentOS-6 - Contrib 0
epel Extra Packages for Enterprise Linux 6 - 
x86_64                                          12,501
extras                               CentOS-6 - Extras 33
remi-php72 Remi's PHP 7.2 RPM repository for Enterprise Linux 6 - 
x86_64                              310
remi-safe Safe Remi's RPM repository for Enterprise Linux 6 - 
x86_64                               2,724
updates                              CentOS-6 - Updates 233

 From another site I got a tip.  I then did "yum clean all"
and     rm -rf /var/cache/yum/epel
           rm -rf /var/cache/yum/x86_64/6/epel

Then, "yum update" worked.  PHP7.2 installed.

Thanks for tips

Remi Collet wrote on 12/4/18 6:32 AM:

**Comment by craigahumphreys** [comment]
Turned out my problem was solved with the following (on CentOS 6.10 x64)

- rm -rf /var/cache/yum/epel
- rm -rf /var/cache/yum/x86_64/6/epel
- yum clean all

Then "yum update" properly installed PHP7.2
